wallet/src/main/scala/org/bitcoins/wallet/models/SpendingInfoTable.scala

```diff
@@ -167,6 +167,14 @@ sealed trait SpendingInfoDb extends DbRowAutoInc[SpendingInfoDb] {
   /** Updates the `blockHash` field */
   def copyWithBlockHash(blockHash: DoubleSha256DigestBE): SpendingInfoType
 
+  state match {
```

Comment:
can you move this to the top? It's confusing to me to find it in the middle of method defintions. Also can you enumerate the `TxoState` again in the pattern match so we can get an exhaustiveness check by the compiler?